pkg/products/marin3r/rateLimitService.go

```diff
@@ -146,29 +147,37 @@ func (r *RateLimitServiceReconciler) reconcileDeployment(ctx context.Context, cl
 
 	deployment := &appsv1.Deployment{
 		ObjectMeta: v1.ObjectMeta{
-			Name:      "ratelimit",
+			Name:      quota.RateLimitName,
 			Namespace: r.Namespace,
 		},
 	}
 
+	key, err := k8sclient.ObjectKeyFromObject(deployment)
+	if err != nil {
+		return integreatlyv1alpha1.PhaseFailed, err
+	}
```

Comment:
If I am looking at the correct source code for `ObjectKeyFromObject()`,
then we don't need any error checking as it never returns errors:
https://github.com/kubernetes-sigs/controller-runtime/blob/b2c90ab82af89fb84120108af14f4ade9df5d787/pkg/client/interfaces.go#L33-L36
And this applies to other `k8sclient.ObjectKeyFromObject()` calls in this PR...

Reply:
Please ignore, I was looking at newer version of controller-runtime.